pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -328,7 +336,17 @@ pub mod pallet {
 			origin: OriginFor<T>,
 			msg: BoundedVec<u8, T::MaxIncomingMessageSize>,
 		) -> DispatchResult {
-			let (domain_address, incoming_msg) = match T::LocalEVMOrigin::ensure_origin(origin)? {
+			Self::deposit_event(Event::IncomingMessageReceived {
+				message: msg.to_vec(),
+			});
+
+			let origin = T::LocalEVMOrigin::ensure_origin(origin).map_err(|e| {
+				Self::deposit_event(Event::InvalidIncomingMessageOrigin);
```

Comment:
We will never see that. Upon error the state will be discarded. We would need to make this method `non_transactional` in order to see events of failures.

Reply:
Ah ok, but making it `non-transactional` would not be OK given that we're calling the LP pallet right?

I can also remove this, the one thing that is important here would be the test that confirms that an incoming xcm call is executed accordingly.

Reply:
> Ah ok, but making it non-transactional would not be OK given that we're calling the LP pallet right?

If we removed the `transactional` component, we would have to ensure no storage is mutated until errors cannot occur anymore. However AFAIK, we can have transactional sub-scope within `process_msg` such as `submit`.

Reply:
The inbound queue implemented by the LP pallet is marked as transactional, does that mean that we can mark `process_msg` as non-transactional then?

https://github.com/centrifuge/centrifuge-chain/blob/main/pallets/liquidity-pools/src/lib.rs#L937

Reply:
> The inbound queue implemented by the LP pallet is marked as transactional, does that mean that we can mark `process_msg` as non-transactional then?

Yes, that would be possible. However, we should keep extrinsics `transactional` and rather check, whether we can remove that layer from the `InboundQueue::submit` impl of `pallet-liquidity-pools`. 

I had to ensure myself via some local tests: If a transactional call `c` executes nested non-transactional calls `nested_c` which mutate storage, then all storage mutations of `c` including `nested_c` are still rolled back if any error is thrown by `c`. 

Transactional layers [come with minor overhead](https://substrate.stackexchange.com/questions/1184/whats-the-overhead-associated-with-the-transactional-macro):
* 50 ns for the layer itself
* ~146 ns writing storage item from `transactional` layer to db which is only ~0.15% overhead compared to weight for one `RocksDb` write